src/shared/utilities/messages.ts

```diff
@@ -4,20 +4,24 @@
  */
 
 import * as vscode from 'vscode'
-import { getIdeProperties, isCloud9 } from '../extensionUtilities'
+import { getIdeProperties, isCloud9, isCn } from '../extensionUtilities'
 import { getLogger, showLogOutputChannel } from '../../shared/logger'
 import { localize } from '../../shared/utilities/vsCodeUtils'
 import { Window } from '../../shared/vscode/window'
 import { ext } from '../extensionGlobals'
 import { Timeout } from './timeoutUtils'
 
-const commandName = localize('AWS.command.viewLogs', 'View AWS Toolkit Logs')
+// function instead of constant to prevent isCn() from running prior to compute region being determined
+// two localized strings instead of a single one with a parameter since this is also used as a command name
+function commandName(): string {
+    return isCn() ? localize('AWS.command.viewLogs.cn', 'View Amazon Toolkit Logs') :  localize('AWS.command.viewLogs', 'View AWS Toolkit Logs')
+}
 
-export function makeCheckLogsMessage(): string {
+export async function makeCheckLogsMessage(): Promise<string> {
```

Comment:
why change to async?

Reply:
Will revert, first iteration contacted metadata service via getIdeProperties.